src/SearchLayout.js

```diff
@@ -11,9 +17,15 @@ let styles = mode => ({
     mode === 'basic' ? 'minmax(250px, 400px) minmax(0, 1fr)' : 'minmax(0, 1fr)',
 })
 
-let SearchLayout = ({ mode, style, ...props }) => (
-  <div style={{ ...styles(mode), ...style }} {...props} />
-)
+let SearchLayout = ({ style, mode, setMode, ...props }) => {
+  let [stateLayout] = React.useState(observable.box('basic'))
+  let layout = mode ? F.stateLens([mode, setMode]) : stateLayout
```

Comment:
What's the observable for? I suspect I'm missing something, but this one-liner seems to work just as well:

```es6
let layout = F.stateLens(mode ? [mode, setMode] : React.useState('basic'))
```

Reply:
Hmm yeah you're right we don't need the observable. Good catch :+1: 

FYI: Using hooks conditionally like that is not recommended: https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level